### PR TITLE
be more permissive with pragmas

### PR DESCRIPTION
**Pull request by glommer**
Especially read only, just let them go!

### CODE REVIEW

File: sqld/src/query_analysis.rs [hunk 1]
```diff
@@ -62,20 +62,34 @@ impl StmtKind {
...
         }
     }
 }
 
+fn is_ro_pragma(name: &QualifiedName, _body: Option<&PragmaBody>) -> bool {
+    matches!(name, QualifiedName {
+            db_name: None,
+            name,
+            alias: None,
+        } if name.0.starts_with("index_")|| name.0 == "encoding" || name.0 == "function_list" || name.0 == "module_list" || name.0.starts_with("table_"))
```
Comment (psarna) [inline sqld/src/query_analysis.rs:84]: Some pragmas are only read-only if they don't have parameters, e.g. `encoding`:
```
PRAGMA encoding;
PRAGMA encoding = 'UTF-8';
PRAGMA encoding = 'UTF-16';
PRAGMA encoding = 'UTF-16le';
PRAGMA encoding = 'UTF-16be';
```

I think we can still just merge as is and fix it in a follow-up by inspecting the `body: Option<&PragmaBody>` parameter, but I'll leave the finall call to @MarinPostma

**Review by psarna — APPROVED** [review]

File: sqld/src/query_analysis.rs (hunk 1 above)
Comment (glommer) [reply sqld/src/query_analysis.rs:84]: or we could whitelist encoding as write. The others are mostly list of stuff, and should be ro.

Comment (psarna) [reply sqld/src/query_analysis.rs:84]: I need to test, but I think it's as simple as: body is Some => write, body is None => read

Comment (psarna) [reply sqld/src/query_analysis.rs:84]: yup, exactly:
pragma encoding = 'utf8':
```rust
name: QualifiedName { db_name: None, name: Name("encoding"), alias: None }, body Some(Equals(Name(Name("utf8"))))
```
pragma encoding:
```rust
name: QualifiedName { db_name: None, name: Name("encoding"), alias: None }, body None
```

**Comment by psarna** [comment]
@glommer never mind about `encoding` actually, it says:
```
The second through fifth forms of this pragma set the encoding that the main database will be created with if it is created by this session.
```

So while technically we should do something like this to only allow read-only encoding pragmas:
```rust
-fn is_ro_pragma(name: &QualifiedName, _body: Option<&PragmaBody>) -> bool {
+fn is_ro_pragma(name: &QualifiedName, body: Option<&PragmaBody>) -> bool {
     matches!(name, QualifiedName {
             db_name: None,
             name,
             alias: None,
-        } if name.0.starts_with("index_")|| name.0 == "encoding" || name.0 == "function_list" || name.0 == "module_list" || name.0.starts_with("table_"))
+        } if name.0.starts_with("index_")|| (name.0 == "encoding" && body.is_none()) || name.0 == "function_list" || name.0 == "module_list" || name.0.starts_with("table_"))
 }
```,

in practice it's meaningless, because in sqld the database is **always** already created anyway, so all encoding pragmas are **effectively** read-only

**Comment by psarna** [comment]
let me just go ahead and merge, and we can always figure out the rough edges later today

**Comment by psarna** [comment]
bors merge

**Comment by glommer** [comment]
right, that's what I meant. For encoding in particular, might as well add it to the list
